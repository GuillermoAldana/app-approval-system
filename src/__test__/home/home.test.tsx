import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import HomePage from '@/app/home';

// Mock del router
const pushMock = jest.fn();
jest.mock('next/navigation', () => ({
    useRouter: jest.fn(() => ({ push: pushMock })),
}));

jest.mock('@/components/Request/Content/Status', () =>
    () => React.createElement('div', null, 'StatusCard')
);
jest.mock('@/components/Request/Content/Table', () =>
    () => React.createElement('div', null, 'TableRequest')
);

describe('<HomePage />', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });
    it('Should be able to render correctly', () => {
        const { asFragment } = render(<HomePage />);
        expect(asFragment()).toMatchSnapshot();
    })
    it('Should be able to renders heading and description', () => {
        render(<HomePage />);
        expect(screen.getByText(/Sistema de Aprobación/i)).toBeInTheDocument();
        expect(
            screen.getByText(
                /Gestiona y supervisa todas las solicitudes de aprobación/i
            )
        ).toBeInTheDocument();
    });

    it('Should be able to renders StatusCard and TableRequest', () => {
        render(<HomePage />);
        expect(screen.getByText(/StatusCard/i)).toBeInTheDocument();
        expect(screen.getByText(/TableRequest/i)).toBeInTheDocument();
    });

    it('Should be able to navigates to /approvals when clicking the create button', () => {
        render(<HomePage />);
        const button = screen.getByText(/\+ Crear una nueva solicitud/i);
        fireEvent.click(button);
        expect(pushMock).toHaveBeenCalledWith('/approvals');
    });
});
