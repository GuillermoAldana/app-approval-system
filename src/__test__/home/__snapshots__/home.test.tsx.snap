// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<HomePage /> Should be able to render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-wjsns7-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-q5srq4-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1ugojh5-MuiTypography-root"
      >
        Sistema de Aprobación
      </h4>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1l6xtz1-MuiTypography-root"
      >
        Gestiona y supervisa todas las solicitudes de aprobación de manera eficiente.
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1bka3le-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        + Crear una nueva solicitud
      </button>
    </div>
  </div>
  <div>
    StatusCard
  </div>
  <div>
    TableRequest
  </div>
</DocumentFragment>
`;
